src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
The time it takes to implement isn't the issue here. We don't know what other consequences an implementation will have, which is why we test. The change currently made is relatively limited in scope, and can be relatively reliably unit tested. Change the mechanism above it, while possibly trivial in implementation, could have further repercussions that we might not catch. Seeing as this is to fix an exception case, it's safer to make the minimal required change with as limited a scope as possible in an easily testable area, and then revisit making a larger change when there is time to thoroughly test it.